pkg/platform/updog/updog.go

```diff
@@ -60,6 +62,11 @@ func (e *executable) runOk(cmd *exec.Cmd) (bool, error) {
 	}
 	err := cmd.Wait()
 	if err != nil {
+		// Updog returns an non zero exit code if there are no available updates.
+		// Don't treat it as an error, instead indicate that there's no available update.
+		if len(cmd.Args) > 1 && cmd.Args[1] == "check-update" && buf.String() == "No update available\n" {
```

Comment:
I think that we might want to handle this with `$1 == "check-update" && exitCode == 1`, or something similar instead of reading into the (human oriented) error message.

`updog`'s ["no update available"](https://github.com/bottlerocket-os/bottlerocket/blob/a07d60c6e540e96392a73a311394b90cb391f39c/sources/updater/updog/src/main.rs#L501) and ["no-update in node's wave"](https://github.com/bottlerocket-os/bottlerocket/blob/a07d60c6e540e96392a73a311394b90cb391f39c/sources/updater/updog/src/main.rs#L506) "errors" are only distinguished by the [printed error string](https://github.com/bottlerocket-os/bottlerocket/blob/a07d60c6e540e96392a73a311394b90cb391f39c/sources/updater/updog/src/error.rs#L206-L213). The update operator, for what its concerned with here anyway, doesn't make a distinction between these two - and I don't think it'll need to (not before the interface is formalized anyway). So, if the agent ran `updog check-update` and it gets a `exitCode == 1` back, then we know that `updog` effectively is reporting "no updates [yet]!".

Reply:
I don't follow how we know that.  How do we know that updog didn't fail to check, in which case brupop should fail as well?

Reply:
I think with the context used as-is that we'll see the same error printed: [`errors::UpdateNotAvailable`](https://github.com/bottlerocket-os/bottlerocket/blob/a07d60c6e540e96392a73a311394b90cb391f39c/sources/updater/updog/src/main.rs#L501). There could be inner error handling I missed or misread that makes this not true! Though in that case, I wonder what we'd see if errors *don't* bubble up. Does it show up by have the error result massaged into `None` (in the check `fn`) from the downstream errors?

Reply:
I'm glossing over the *earlier* errors. My bad.

Reply:
I would really like to see different error codes, but I know this isn't the API we want to formalize so its unlikely we'll get any additional distinguishers there. Without some method to distinguish between operational error and check-result "errors", we're having to resort to checking strings..and that's no bueno.